### PR TITLE
REP-4928 Keystone v2 Filter Should Return 401 For Self-Validating Tokens

### PR DESCRIPTION
**Pull request by dmnjohns**

### CODE REVIEW

File: repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/keystonev2/selfvalidation/KeystoneV2SelfValidationTest.groovy [hunk 1]
```diff
@@ -54,7 +57,10 @@ class KeystoneV2SelfValidationTest extends ReposeValveTest {
     }
 
     def setup() {
+        fakeIdentityV2Service.resetCounts()
         fakeIdentityV2Service.resetDefaultParameters()
+        fakeIdentityV2Service.resetHandlers()
+        sleep(500)
```
Comment (adrianjgeorge) [inline repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/keystonev2/selfvalidation/KeystoneV2SelfValidationTest.groovy]: ???

Comment (dmnjohns) [inline repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/keystonev2/selfvalidation/KeystoneV2SelfValidationTest.groovy]: Inelegantly avoiding the Akka cache.

Comment (Mario-Lopez) [inline repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/keystonev2/selfvalidation/KeystoneV2SelfValidationTest.groovy]: ⚡️

Comment (adrianjgeorge) [inline repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/keystonev2/selfvalidation/KeystoneV2SelfValidationTest.groovy]: is that needed here, where we've done work related to the cache?

Comment (dmnjohns) [inline repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/keystonev2/selfvalidation/KeystoneV2SelfValidationTest.groovy]: You mean in Repose 8? I believe it's still needed. If I remember correctly, the work we did was to bypass the Akka cache on retries (like when the admin token expires). In this case, it's not a retry for a single request, but rather behavior across multiple requests, so we don't hit the code that bypasses the cache.

Comment (adrianjgeorge) [inline repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/keystonev2/selfvalidation/KeystoneV2SelfValidationTest.groovy]: Ugh.

Comment (adrianjgeorge) [inline repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/keystonev2/selfvalidation/KeystoneV2SelfValidationTest.groovy]: Man i really want to make that configurable.

Comment (dmnjohns) [inline repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/keystonev2/selfvalidation/KeystoneV2SelfValidationTest.groovy]: I am 100% with you on that one.

Comment (Mario-Lopez) [inline repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/keystonev2/selfvalidation/KeystoneV2SelfValidationTest.groovy]: We have like 3 duplicate stories for this:
https://repose.atlassian.net/browse/REP-4457

**Pull request merged**